phase1/gce/configure-vm-kubeadm.sh

```diff
@@ -3,30 +3,52 @@
 TOKEN=$(get_metadata "k8s-kubeadm-token")
 KUBEADM_VERSION=$(get_metadata "k8s-kubeadm-version")
 KUBERNETES_VERSION=$(get_metadata "k8s-kubernetes-version")
-
+KUBELET_VERSION=$(get_metadata "k8s-kubelet-version")
 curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add -
 
-if [[ "${KUBEADM_VERSION}" == stable ]]; then
+if [[ "${KUBELET_VERSION}" == stable ]]; then
   cat <<EOF > /etc/apt/sources.list.d/kubernetes.list
   deb http://apt.kubernetes.io/ kubernetes-xenial main
 EOF
-
   apt-get update
+  # kubeadm is installed with the kubelet so that the 
+  # kubelet has the configuration at a matching version
   apt-get install -y kubelet kubeadm kubectl kubernetes-cni
-elif [[ "${KUBEADM_VERSION}" == "gs://"* ]]; then
+elif [[ "${KUBELET_VERSION}" == "gs://"* ]]; then
   TMPDIR=/tmp/k8s-debs
   mkdir $TMPDIR
-  gsutil rsync "${KUBEADM_VERSION}" $TMPDIR
+  gsutil rsync "${KUBELET_VERSION}" $TMPDIR
+  # kubeadm is installed with the kubelet so that the 
+  # kubelet has the configuration at a matching version
   dpkg -i $TMPDIR/{kubelet,kubeadm,kubectl,kubernetes-cni}.deb || echo Ignoring expected dpkg failure
   apt-get install -f -y
   systemctl enable kubelet
   systemctl start kubelet
   rm -rf $TMPDIR
 else
-  echo "Don't know how to handle version: $KUBEADM_VERSION"
+  echo "Don't know how to handle kubelet version: $KUBELET_VERSION"
   exit 1
 fi
 
+if [[ "${KUBEADM_VERSION}" != "${KUBELET_VERSION}" ]]; then
+  if [[ "${KUBEADM_VERSION}" == stable ]]; then
+    # Cannot install packages as they will update the 
+    # kubelet configuration to a version that does not match
+    # the installed kubelet
+    echo "Kubeadm version of 'stable' is not supported with kubelet version that is not also 'stable'."
+    exit 1
+  elif [[ "${KUBEADM_VERSION}" == "gs://"* ]]; then
+    TMPDIR=/tmp/k8s-debs
+    mkdir $TMPDIR
+    gsutil cp "${KUBEADM_VERSION}/kubeadm" $TMPDIR/kubeadm
+    cat $TMPDIR/kubeadm > /usr/bin/kubeadm
```

Comment:
I think you're looking for the `cp` command.

Reply:
Any particular advantage of using cp instead of gsutil cp to full from a gs url?

Reply:
I should be more specific:

    cat $TMPDIR/kubeadm > /usr/bin/kubeadm
 
could just be

    cp $TMPDIR/kubeadm /usr/bin/kubeadm

Reply:
In fact, instead of using `gsutil cp <source> /tmp/kubeadm` and then `cp /tmp/kubeadm /usr/bin/kubeadm`, this could just be one line:

    gsutil cp "${KUBEADM_VERSION}/kubeadm" /usr/bin/kubeadm